src/modules/fancyzones/editor/FancyZonesEditor/Models/LayoutModel.cs

```diff
@@ -103,107 +120,165 @@ protected virtual void FirePropertyChanged(string propertyName)
         // Removes this Layout from the registry and the loaded CustomModels list
         public void Delete()
         {
-            RegistryKey key = Registry.CurrentUser.OpenSubKey(_registryPath, true);
-            if (key != null)
-            {
-                key.DeleteValue(Name);
-            }
 
             int i = _customModels.IndexOf(this);
             if (i != -1)
             {
                 _customModels.RemoveAt(i);
+                _deletedCustomModels.Add(Guid.ToString().ToUpper());
+            }
+        }
+
+        public static void SerializeDeletedCustomZoneSets()
+        {
+            FileStream outputStream = File.Open(Settings.CustomZoneSetsTmpFile, FileMode.Create);
+            var writer = new Utf8JsonWriter(outputStream, options: default);
+            writer.WriteStartObject();
+            writer.WriteStartArray("deleted-custom-zone-sets");
+            foreach (string zoneSet in _deletedCustomModels)
+            {
+                writer.WriteStringValue(zoneSet);
             }
+
+            writer.WriteEndArray();
+            writer.WriteEndObject();
+            writer.Flush();
+            outputStream.Close();
         }
 
         // Loads all the Layouts persisted under the Layouts key in the registry
         public static ObservableCollection<LayoutModel> LoadCustomModels()
         {
             _customModels = new ObservableCollection<LayoutModel>();
 
-            RegistryKey key = Registry.CurrentUser.OpenSubKey(_registryPath);
-            if (key != null)
+            FileStream inputStream = File.Open(Settings.CustomZoneSetsTmpFile, FileMode.Open);

```

Comment:
Comment for `LoadCustomModels` is obsolete. Also, Is the `Settings.CustomZoneSetsTmpFile` always set at this point?